sourcepolicy/engine_test.go

```diff
@@ -0,0 +1,354 @@
+package sourcepolicy
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/moby/buildkit/solver/pb"
+	spb "github.com/moby/buildkit/sourcepolicy/pb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEngineEvaluate(t *testing.T) {
+	t.Run("Deny All", testDenyAll)
+	t.Run("Allow Deny", testAllowDeny)
+	t.Run("Convert", testConvert)
+	t.Run("Convert Deny", testConvertDeny)
+	t.Run("Allow Convert Deny", testAllowConvertDeny)
+	t.Run("Test convert loop", testConvertLoop)
+	t.Run("Test convert http", testConvertHTTP)
+	t.Run("Test convert regex", testConvertRegex)
+}
+
+func testConvertRegex(t *testing.T) {
+	pol := &spb.Policy{
+		Rules: []*spb.Rule{
+			{
+				Action: spb.PolicyAction_CONVERT,
+				Source: &spb.Source{
+					Type:       "docker-image",
+					Identifier: `docker\.io/library/golang:(.*)`,
+					MatchType:  spb.MatchType_REGEX,
+				},
+				Destination: &spb.Destination{
+					Identifier: "docker-image://fakereg.io/library/golang:${1}",
+				},
```

Comment:
I might be missing something - why is `Source.Identifier` just the ref, while `Destination.Identifier` is the `Type` and the ref?

If we can, I think the source and destination structs should be as similar as possible unless we have a good reason for keeping them different.